File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/prism/PrismFightAttackerRemoveMessage.scala





















// Generated on 05/03/2013 10:34:59
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PrismFightAttackerRemoveMessage extends MessageDeserializer {
  val messageId = 5897

  protected trait Message extends BaseMessage with Serializable {
    val messageId = PrismFightAttackerRemoveMessage.messageId

    val fightId: Double
    val fighterToRemoveId: Int

    def serialize(buf: Buffer) {
      buf.writeDouble(fightId)
      buf.writeInt(fighterToRemoveId)
    }

  }

  protected def create(fightId: Double, fighterToRemoveId: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val fightId = buf.readDouble()
    val fighterToRemoveId = buf.readInt()

    create(fightId, fighterToRemoveId)
  }

}

object PrismFightAttackerRemoveMessage extends PrismFightAttackerRemoveMessage {
  case class MessageImpl(fightId: Double, fighterToRemoveId: Int) extends Message
  type Target = MessageImpl

  override def create(fightId: Double, fighterToRemoveId: Int) = MessageImpl(fightId, fighterToRemoveId)

  def apply(fightId: Double, fighterToRemoveId: Int) = create(fightId, fighterToRemoveId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PrismFightAttackerRemoveMessage.java


// Generated on 05/08/2013 19:37:57
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PrismFightAttackerRemoveMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5897;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public double fightId;
    public int fighterToRemoveId;
    
    public PrismFightAttackerRemoveMessage() { }
    
    public PrismFightAttackerRemoveMessage(double fightId, int fighterToRemoveId) {
        this.fightId = fightId;
        this.fighterToRemoveId = fighterToRemoveId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeDouble(fightId);
        buf.writeInt(fighterToRemoveId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        fightId = buf.readDouble();
        fighterToRemoveId = buf.readInt();
        if (fighterToRemoveId < 0)
            throw new RuntimeException("Forbidden value on fighterToRemoveId = " + fighterToRemoveId + ", it doesn't respect the following condition : fighterToRemoveId < 0");
    }
    
}
